fms/include/fms.inc

```diff
@@ -16,729 +16,26 @@
-          ELSE IF ( nml_iostats(2) .EQ. 0 .AND. nml_iostats(3) .NE.0 ) THEN
-             nml_iostats(2) = nml_iostats(3)
-          ELSE
-             nml_iostats(2) = nml_iostats(4)
-             nml_iostats(2) = nml_iostats(4)
-          END IF
-       END IF
-    END IF
-
-    ! Broadcast nml_errors
-    CALL mpp_broadcast(nml_iostats,5,mpp_root_pe())
-    nml_errors%multipleNMLSinFile = nml_iostats(1)
-    nml_errors%badType1 = nml_iostats(2)
-    nml_errors%badType2 = nml_iostats(3)
-    nml_errors%missingVar = nml_iostats(4)
-    nml_errors%NotInFile = nml_iostats(5)
-
-    do_nml_error_init = .FALSE.
-  END SUBROUTINE nml_error_init
-
-!#######################################################################
-
-!> @brief match the input character string to a string
-!!     in an array/list of character strings
-!! @return If an exact match was found then true is returned, otherwise false is returned.
-!! @details Tries to find a match for a character string in a list of character strings.
-!!      The match is case sensitive and disregards blank characters to the right of
-!!      the string.
-!!
-!!      <br>Examples<br>
-!!      @code{.F90}
-!!       string = "def"
-!!       string_array = (/ "abcd", "def ", "fghi" /)
-!!
-!!       string_array_index ( string, string_array, index )
-!!      @endcode
-!!       Returns: TRUE, index = 2
-function string_array_index ( string, string_array, index ) result (found)
-character(len=*),  intent(in)  :: string !< Character string of arbitrary length.
-character(len=*),  intent(in)  :: string_array(:) !< Array/list of character strings.
-integer, optional, intent(out) :: index !< The index of string_array where the first match was found. If
-                                        !! no match was found then index = 0.
-logical :: found !< If an exact match was found then TRUE is returned, otherwise FALSE is returned.
-integer :: i
-
-! initialize this function to false
-! loop thru string_array and exit when a match is found
-
-  found = .false.
-  if (present(index)) index = 0
-
-  do i = 1, size(string_array(:))
-    ! found a string match ?
-    if ( trim(string) == trim(string_array(i)) ) then
-         found = .true.
-         if (present(index)) index = i
-         exit
-    endif
-  enddo
-
-end function string_array_index
-
-!#######################################################################
 
 !> @brief Determines if a real input array has monotonically increasing or
 !!     decreasing values.
 !! @return If the input array of real values either increases or decreases monotonically then true
 !! is returned, otherwise false is returned.
-function monotonic_array ( array, direction )
-real,    intent(in)            :: array(:) !< An array of real values. If the size(array) < 2 this function
-                                           !! assumes the array is not monotonic, no fatal error will occur.
+
+function MONOTONIC_ARRAY_(array, direction) result(ret)
```

Comment:
I guess this routine doesn't take into account elements having the same value, e.g. -2, -1, 0, 0, 1, 2

Reply:
Need to get my math concepts straight.  Monotonic increasing/decreasing = strictly increasing/decreasing and not remaining constant. All's good.

Reply:
I believe monotonic functions, as commonly defined, can have plateaus of constant value. So this function seems to use a peculiar definition of "monotonic".

Reply:
There is a difference between a monotonic and a  _strictly_ monotonic function.  If this function only enforces (X<sub>1</sub> &lt; X<sub>2</sub>), then it is testing for _strict_ increasing monotonicity.  If it is (X<sub>1</sub> &gt; X<sub>2</sub>), then it is enforcing _strict_ decreasing monotonicity.

Reply:
Perhaps it's worth commenting in the code that monotonic refers to strictly monotonic.

Reply:
0b397a624e7b5dd2ea825372a8f1ff66029b63fc